src/kdbindings/node.h

```diff
@@ -206,15 +210,21 @@ class PropertyNode : public NodeInterface<PropertyType>
     const bool *dirtyVariable() const override { return &m_dirty; }
 
 private:
-    void setProperty(Property<PropertyType> &property)
+    void setProperty(PropertyReference property)
     {
         m_property = &property;
-        m_valueChangedHandle = m_property->valueChanged().connect(&PropertyNode<PropertyType>::markDirty, this);
-        m_movedHandle = m_property->m_moved.connect(&PropertyNode<PropertyType>::propertyMoved, this);
-        m_destroyedHandle = m_property->destroyed().connect(&PropertyNode<PropertyType>::propertyDestroyed, this);
+
+        // Connect to the valueChanged signal in all cases
+        m_valueChangedHandle = m_property->valueChanged().connect([this]() { this->markDirty(); });
+
+        // Only connect move and destruction signals if property is non-const
+        if constexpr (!IsConst) {
```

Comment:
This is not correct, both of these signals need to be connected, even if the property is const!

Otherwise, we will not be able to follow the property when it is moved and may even try to access it after it has been destroyed!

You may have to make the m_moved signal mutable as well, but that shouldn't be an issue.